module/spl/spl-taskq.c

```diff
@@ -1081,6 +1108,10 @@ taskq_destroy(taskq_t *tq)
 		task_free(tq, t);
 	}
 
+	down_write(&tq_list_sem);
+	list_del(&tq->tq_taskqs);
+	up_write(&tq_list_sem);
```

Comment:
I'd also recommend removing it from the list after `taskq_wait()` above and before the spin lock.  We really don't want the taskq_t to be findable while it's being destroyed.